### ISSUE TITLE
[cmd/flag]: validation is skipping empty flag value

### ISSUE THREAD

**Issue by Bidon15** [celestiaorg/celestia-node#3003]
### Celestia Node version

v0.12.1

### OS

linux

### Steps to reproduce it

```bash
celestia light start --p2p.network= --node.store=/home/celestia/.celestia-light-mocha-4 --core.ip=full.consensus.mocha-4.celestia-mocha.com --gateway --da.grpc.namespace=000008e5f679bf7116cb --da.grpc.listen=0.0.0.0:26650
```

### Expected result

start fails with error that `p2p.network` is failed due to empty value provided

### Actual result

celestia mainnet is started

**Comment by Wondertan** [comment]
> start --p2p.network= --node.

The empty network is actually provided. Doesn't look like a problem.

**Comment by Bidon15** [comment]
this is a UX problem that when a user is providing emptiness, node needs to reject such behaviours and ask for a specific value instead of accepting emptiness. 

Emptiness is fine when there is not flag at all. 

The current bug displays a UX problem when a user expects the light node to join mocha testnet, but ended up in celestia mainnet

**Comment by Wondertan** [comment]
I misunderstood the issue due to reading the following as Actual Result, rather than Expected and I thought you wanted to make it work with empty.

> start fails with error that p2p.network is failed due to empty value provided

**Comment by sontrinh16** [comment]
hmmm i believe because of this [line](https://github.com/celestiaorg/celestia-node/blob/main/nodebuilder/p2p/flags.go#L76). We can return an error here instead if we want to reject it

**Comment by 00x-dx** [comment]
I have pushed a fix at #3004. Ideally a shorthand network notation is good to have.
